src/benchmarklib/benchmark_runner.cpp

```diff
@@ -483,7 +485,10 @@ void BenchmarkRunner::_create_report(std::ostream& stream) const {
 std::shared_ptr<SQLPipeline> BenchmarkRunner::_build_sql_pipeline(const QueryID query_id) const {
   // Create an SQLPipeline for this query
   const auto sql = _query_generator->build_query(query_id);
-  auto pipeline_builder = SQLPipelineBuilder{sql}.with_mvcc(_config.use_mvcc);
+  auto pipeline_builder = SQLPipelineBuilder{sql}
+                              .with_sql_lqp_cache(_sql_lqp_cache)
```

Comment:
I don't like having to pass the cache in if we want to use it. The pipeline is supposed to be a single, one-stop solution for executing SQL queries. We shouldn't make it any more difficult than absolutely necessary. Even if the original issue called for non-global caches, I'd suggest to have a default cache in the pipeline (or the pipeline builder) and being able to overwrite/disable it using `with_sql_lqp_cache`.

Reply:
And what cache would that be? A global singleton?

Reply:
No, just a default cache stored as a static member in the pipeline (builder).

Maybe we should decide what we want to fix: Do we want the possibility of using different (or no) caches or do we want to get rid of the default cache?

For me, the first part is more important. Making the benchmark runner (and others) aware of the cache here means that we will have to touch multiple places when we want to disable the cache by default.

Reply:
> Maybe we should decide what we want to fix: Do we want the possibility of using different (or no) caches or do we want to get rid of the default cache?

Both, see below.

> No, just a default cache stored as a static member in the pipeline (builder).

That *is* a global by definition, right? (I don't see how you could have a sensible default cache without it being a global.)

Caching can have unexpected effects like: the Optimizer is not invoked (e.g., chunk pruning info is not updated, the rule you're programming/testing not getting invoked), the SQLTranslator is not invoked, the LQPTranslator is not invoked, caching costs. I'd say caching should require a conscious decision on the libhyrise-user side. 

> Making the benchmark runner (and others) aware of the cache here means that we will have to touch multiple places when we want to disable the cache by default.

As shown by the diff of this PR, every explicit assignment of a cache was done for a place where an explicit cache was used anyway, except for the BenchmarkRunner (where, with the introduction of TPCC, we have to make a cache/no-cache decision in the future) and the Console. I don't see how this diff would be significantly shorter with a default cache. But it's likely I misunderstand your point,

Reply:
Actually, the global cache is a pain: Every test using SQL then has to make sure this cache is empty before doing anything. There were a few seemingly random places where `SQLPhysicalPlanCache::get().clear()` was called that I could thankfully remove.

Reply:
Let's discuss this on Wednesday.